### ISSUE TITLE
Should parameters use their method accessibility when applying naming styles?

### ISSUE THREAD

**Issue by jasonmalinowski** [dotnet/roslyn#42037]
Right now when we compute the applicable naming styles for a parameter, the parameter picks up the accessibility of the containing method. This is _potentially_ useful, as it means you could have a rule that says all parameters that are on a public API must meet some standard style, but you are otherwise inconsistent in the internal parts of your codebase. This however is also confusing as the submitter of [this feedback item](https://developercommunity.visualstudio.com/content/problem/802550/visual-studio-community-2017-and-2019-naming-code.html) discovered.

Does this behavior make sense? Should we just make parameters ignore accessibility? Minimally, should the naming styles symbol specification dialog as shown by that customer have given a warning they were creating a rule that wouldn't do anything?

**Comment by kendrahavens** [comment]
Design meeting notes: 
There should be no distinction in accessibility in naming styles. You can't put public or private before a parameter so it doesn't have accessibility.

**Comment by sharwell** [comment]
I'm bringing this back to design review for this week, because the last meeting overlooked the use case that led to the current design:

While it is possible to write a naming style that enforces naming rules only for public types and members, it is not possible (without the distinction described above) to write a naming rule that only enforces parameter names for publicly-visible parameters.

**Comment by jasonmalinowski** [comment]
@sharwell That was discussed in the meeting somewhat, we didn't really have a motivating scenario for it. (In fact we found any scenarios we came up with particularly dismotivating.) If we missed one then do bring that up.

**Comment by vatsalyaagrawal** [comment]
Design meeting notes:
There should be no distinction in accessibility in naming styles. You can't put public or private before a parameter so it doesn't have accessibility. 

Note: Conclusion is same as last time.

**Comment by CyrusNajmabadi** [comment]
Closing out as this was resolved and there was no movement/change since hten.
